### DISCUSSION TITLE
Listing all environment variables that starts with a string

### DISCUSSION THREAD

**Discussion by cronhan** [nushell/nushell#12794] [Q&A] [answered]
Good day, new-ish on-and-off user here.
I did a cursory search for this (here and on Google/DDG) but I could not find an answer, and my knowledge of Nu is not yet what it should be :) So here is my question:
In CMD, I can do the following:
```bat
> set a
ALLUSERSPROFILE=C:\ProgramData
APPDATA=C:\Users\user\AppData\Roaming
```
Same thing in PowerShell:
```posh
> Get-ChildItem env:a*
Name                           Value
----                           -----
APPDATA                        C:\Users\user\AppData\Roaming
ALLUSERSPROFILE                C:\ProgramData
```
How can I accomplish the same in nu? Things I have tried that do not work:
```nush
~> $env.a*
Error: nu::shell::column_not_found

~> $env | select a*
Error: nu::shell::column_not_found

~> $env | str starts-with 'a'
Error: nu::shell::only_supports_this_input_type

~> $env | any {|| $in.startswith "a"}
Error: nu::parser::parse_mismatch
```

**Comment by ghost** [comment, marked as answer]
```nu
$env | transpose key value | where ($it.key | str starts-with 'a')
```
since `$env` is a record and not a table we first have to transpose it to one in order to make it properly iterable. -> `transpose key value`

and now we can use `where`, which filters a list/table (and it allows the use of the spicial variable `$it`)

**Comment by cronhan** [comment]
Thank you so much for the detailed explanation! My next task is to make my first function using this knowledge :)
